Error while cross-compiling...

In file included from /home/user/ALPHABOX/alpha-app/app/recorder/main.cpp:27:0:
/home/user/ALPHABOX/alpha-app/app/recorder/backward.hpp:1938:3: warning: #warning ":/ sorry, ain't know no nothing none not of your architecture!"
/home/user/ALPHABOX/alpha-app/app/recorder/backward.hpp: In static member function 'static void backward::SignalHandling::sig_handler(int, siginfo_t*, void*)':
/home/user/ALPHABOX/alpha-app/app/recorder/backward.hpp:1929:15: warning: unused variable 'uctx'
In file included from /home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/bits/move.h:38:0,
                 from /home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/bits/stl_pair.h:60,
                 from /home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/bits/stl_algobase.h:66,
                 from /home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/bits/char_traits.h:41,
                 from /home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/ios:41,
                 from /home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/ostream:40,
                 from /home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/iostream:40,
                 from /home/user/ALPHABOX/alpha-app/app/recorder/main.cpp:4:
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/type_traits: At global scope:
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/type_traits: In instantiation of 'const bool std::__is_convertible_helper<backward::SourceFile, backward::SourceFile, false>::__value':
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/type_traits:312:5:   instantiated from 'std::is_convertible<backward::SourceFile, backward::SourceFile>'
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/bits/stl_map.h:451:11:   instantiated from 'mapped_type& std::map<_Key, _Tp, _Compare, _Alloc>::operator[](const key_type&) [with _Key = std::basic_string<char>, _Tp = backward::SourceFile, _Compare = std::less<std::basic_string<char> >, _Alloc = std::allocator<std::pair<const std::basic_string<char>, backward::SourceFile> >, mapped_type = backward::SourceFile, key_type = std::basic_string<char>]'
/home/user/ALPHABOX/alpha-app/app/recorder/backward.hpp:1647:49:   instantiated from here
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/type_traits:302:71: error: no matching function for call to 'backward::SourceFile::SourceFile(backward::SourceFile)'
/home/user/ALPHABOX/alpha-app/app/recorder/backward.hpp:1481:2: note: candidates are: backward::SourceFile::SourceFile(const std::string&)
/home/user/ALPHABOX/alpha-app/app/recorder/backward.hpp:1480:2: note:                 backward::SourceFile::SourceFile()
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/type_traits:302:71: error:   initializing argument 1 of 'static std::__sfinae_types::__one std::__is_convertible_helper<_From, _To, false>::__test(_To) [with _From = backward::SourceFile, _To = backward::SourceFile, std::__sfinae_types::__one = char]'
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/type_traits: In instantiation of 'std::is_convertible<backward::SourceFile, backward::SourceFile>':
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/bits/stl_map.h:451:11:   instantiated from 'mapped_type& std::map<_Key, _Tp, _Compare, _Alloc>::operator[](const key_type&) [with _Key = std::basic_string<char>, _Tp = backward::SourceFile, _Compare = std::less<std::basic_string<char> >, _Alloc = std::allocator<std::pair<const std::basic_string<char>, backward::SourceFile> >, mapped_type = backward::SourceFile, key_type = std::basic_string<char>]'
/home/user/ALPHABOX/alpha-app/app/recorder/backward.hpp:1647:49:   instantiated from here
/home/user/SDK/linux/linux-current/ti-sdk-am335x-evm-05.04.01.00/linux-devkit/bin/../lib/gcc/arm-arago-linux-gnueabi/4.5.3/../../../../arm-arago-linux-gnueabi/include/c++/4.5.3/type_traits:312:5: error: 'std::__is_convertible_helper<backward::SourceFile, backward::SourceFile, false>::__value' is not a valid template argument for type 'bool' because it is a non-constant expression

--- a/backward.hpp
+++ b/backward.hpp
@@ -1934,6 +1934,8 @@ private:
 		error_addr = reinterpret_cast<void*>(uctx->uc_mcontext.gregs[REG_RIP]);
 #elif defined(REG_EIP) // x86_32
 		error_addr = reinterpret_cast<void*>(uctx->uc_mcontext.gregs[REG_EIP]);
+#elif defined(__arm__)
+		error_addr = reinterpret_cast<void*>(uctx->uc_mcontext.arm_pc);
 #else
 #	warning ":/ sorry, ain't know no nothing none not of your architecture!"
 #endif

